### PR TITLE
Zoom at point adjustments

### PR DESCRIPTION
**Pull request by pklampros**
Various adjustments to zoom in/out processes to make them smoother and remove plan drift. See commit messages for more detail

### CODE REVIEW

File: depthmapX/viewhelpers.cpp [hunk 1]
```diff
@@ -2,13 +2,12 @@
 #include <time.h>
 
 namespace ViewHelpers {
-    QPoint calculateCenter(QPoint point, const QPoint &oldCentre, double factor)
+    Point2f calculateCenter(QPoint& point, const QPoint &oldCentre, double factor)
```
Comment (blsemo) [inline depthmapX/viewhelpers.cpp]: Should const QPoint& point

File: depthmapX/depthmapView.h [hunk 2]
```diff
@@ -178,6 +178,7 @@ public slots:
    void ResetHoverWnd(const QPoint& p = QPoint(-1,-1));
    void CreateHoverWnd();
 
+   void ZoomTowards(double ratio, const Point2f& point);
```
Comment (blsemo) [inline depthmapX/depthmapView.h]: What's the diff between ZoomIn and ZoomTowards? Needs at least to be explained if we need both functions.

**Review by blsemo — COMMENTED** [review]
Looks fine apart from the two comments I put in the review.

**Comment by pklampros** [comment]
Fixed!

**Pull request merged**